riva/clients/asr/client_call.cc

```diff
@@ -13,45 +13,63 @@ ClientCall::ClientCall(uint32_t corr_id, bool word_time_offsets)
   recv_final_flags.reserve(1000);
 }
 
+ClientCall::~ClientCall() {
+  if (pipeline_states_logs_)
+    pipeline_states_logs_.close();
+}
+
 void
 ClientCall::AppendResult(const nr_asr::StreamingRecognitionResult& result)
 {
-  bool is_final = result.is_final();
-  if (latest_result_.final_transcripts.size() < 1) {
-    latest_result_.final_transcripts.resize(1);
-    latest_result_.final_transcripts[0] = "";
-  }
-
-  if (is_final) {
-    int num_alternatives = result.alternatives_size();
-    latest_result_.final_transcripts.resize(num_alternatives);
-    latest_result_.final_scores.resize(num_alternatives);
-    latest_result_.final_time_stamps.resize(num_alternatives);
-    for (int a = 0; a < num_alternatives; ++a) {
-      // Append to transcript
-      latest_result_.final_transcripts[a] += result.alternatives(a).transcript();
-      latest_result_.final_scores[a] += result.alternatives(a).confidence();
+  if (result.has_pipeline_states()) {
+    auto pipeline_states = result.pipeline_states();
+    int prob_states_count = pipeline_states.vad_probabilities_size();
+    std::string vad_log = "";
+    for (int i = 0; i < prob_states_count; i++) {
+      vad_log += std::to_string(pipeline_states.vad_probabilities(i)) + " ";
+    }
+    if(!pipeline_states_logs_){
+      pipeline_states_logs_.open("riva_asr_pipeline_states.log");
```

Comment:
Would it be better if we take output file from user instead of hardcoding it to riva_asr_pipeline_states.log?

Reply:
Idea was to dump the VAD states in a file instead of dumping them on the terminal.

> Would it be better if we take output file from user instead of hardcoding it to riva_asr_pipeline_states.log

This sounds good!!

Reply:
Lets keep it as is. Since this is part of additional custom_config, and is not hit in general code flow. This would be an explicit usecase and if someone using this config, they would know how/where to interpret/consume them.